### PR TITLE
[YUNIKORN-1027] Go: deprecation of io/ioutil

### PR DESCRIPTION
**Pull request by steinsgateted**
### What is this PR for?
Go's ioutil package to be deprecated in 1.16
change ioutil to io or os

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-1027

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/application_graphviz_test.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build graphviz
 // +build graphviz
```
Comment (pbacsko) [inline pkg/scheduler/objects/application_graphviz_test.go:2]: Remove the old tag.

File: pkg/common/configs/config_test.go [hunk 2]
```diff
@@ -239,12 +239,12 @@ func SerdeTest(t *testing.T, conf SchedulerConfig, description string) {
 }
 
 func CreateConfig(data string) (*SchedulerConfig, error) {
-	dir, err := ioutil.TempDir("", "test-scheduler-config")
+	dir, err := os.TempDir("", "test-scheduler-config")
```
Comment (pbacsko) [inline pkg/common/configs/config_test.go:242]: Does this even compile?

It looks like you can use either `os.TempDir()` (no args) or `os.MkdirTemp(dir, pattern string)`. Please check.

**Review by pbacsko — CHANGES_REQUESTED** [review]
Please check `os.TempDir()`, doesn't look correct. Thanks!

File: pkg/scheduler/objects/application_graphviz_test.go (hunk 1 above)
Comment (steinsgateted) [reply pkg/scheduler/objects/application_graphviz_test.go]: Thanks, I removed // +build graphviz

File: pkg/common/configs/config_test.go (hunk 2 above)
Comment (steinsgateted) [reply pkg/common/configs/config_test.go]: sorry i made a mistake. change to : os.MkdirTemp("", "test-scheduler-config")

**Comment by steinsgateted** [comment]
thank you for your review

**Comment by pbacsko** [comment]
We have the same problem as in k8shim, we need both "build" comments...

```Error: level=warning msg="[runner] Can't run linter goanalysis_metalinter: deadcode: analysis skipped: errors in package: [/home/runner/work/incubator-yunikorn-core/incubator-yunikorn-core/pkg/entrypoint/entrypoint.go:30:2: could not import github.com/apache/incubator-yunikorn-core/pkg/scheduler (/home/runner/work/incubator-yunikorn-core/incubator-yunikorn-core/pkg/scheduler/context.go:37:2: could not import github.com/apache/incubator-yunikorn-core/pkg/scheduler/objects (pkg/scheduler/context.go:37:2: //go:build comment without // +build comment)) /home/runner/work/incubator-yunikorn-core/incubator-yunikorn-core/pkg/entrypoint/entrypoint.go:31:2: could not import github.com/apache/incubator-yunikorn-core/pkg/webservice (/home/runner/work/incubator-yunikorn-core/incubator-yunikorn-core/pkg/webservice/handlers.go:41:2: could not import github.com/apache/incubator-yunikorn-core/pkg/scheduler (/home/runner/work/incubator-yunikorn-core/incubator-yunikorn-core/pkg/scheduler/context.go:37:2: could not import github.com/apache/incubator-yunikorn-core/pkg/scheduler/objects (pkg/scheduler/context.go:37:2: //go:build comment without // +build comment)))]"
Error: level=warning msg="[runner] Can't run linter unused: buildir: analysis skipped: errors in package: [pkg/scheduler/context.go:37:2: //go:build comment without // +build comment]"
Error: level=error msg="Running error: buildir: analysis skipped: errors in package: [pkg/scheduler/context.go:37:2: //go:build comment without // +build comment]"
```

**Comment by steinsgateted** [comment]
I modified. Thank you.

**Review by pbacsko — APPROVED** [review]
LGTM +1

pending merge conflict

**Comment by steinsgateted** [comment]
merge conflict. Thank you.

**Comment by pbacsko** [comment]
Thanks @steinsgateted.

**Pull request merged**